### PR TITLE
make invariant errors into error messages

### PR DESCRIPTION
**Pull request by oconnor663**
r? @mlsteele

### CODE REVIEW

File: go/chat/boxer.go [hunk 1]
```diff
@@ -128,6 +128,9 @@ func (b *Boxer) UnboxMessage(ctx context.Context, boxed chat1.MessageBoxed, conv
 	}
 
 	unboxed, ierr := b.unbox(ctx, boxed, encryptionKey)
+	if ierr == nil {
+		ierr = b.checkInvariants(ctx, convID, boxed, unboxed)
```
Comment (oconnor663) [inline go/chat/boxer.go]: Factored out an inner function, like you suggested.

**Comment by oconnor663** [comment]
This will fix https://github.com/keybase/client/issues/6226.

**Comment by mlsteele** [comment]
It might be worth checking if any of the duplicate bodies are the first message in the conv. I have a suspicion that would might cause further gui issues.

**Comment by oconnor663** [comment]
Those would be msgID 1, right? The query result [here](https://keybase.atlassian.net/browse/CORE-4713) suggests that that's not what's happening I think.

**Comment by mlsteele** [comment]
Yes, ok, good.

**Pull request merged**